test/unit/index/test-rangesegment.cpp

```diff
@@ -0,0 +1,84 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hpp"
+#include "gtest/gtest.h"
+
+TEST(RangeSegmentTest, Constructors)
+{
+  RAJA::RangeSegment first(0, 10);
+  RAJA::RangeSegment copied(first);
+
+  ASSERT_EQ(first, copied);
+
+  RAJA::RangeSegment moved(std::move(first));
+
+  ASSERT_EQ(moved, copied);
+
+  RAJA::RangeSegment r1(-10, 7);
```

Comment:
@rhornung67 - I have noticed that constructing a RangeSegment works with *any* indices, even when end < begin. Do we want to enforce something like (begin < end) in the constructor, since we require a `RangeStrideSegment` for negative strides?